consensus/hotstuff/votecollector/base_vote_collector.go

```diff
@@ -0,0 +1,22 @@
+package votecollector
+
+import (
+	"github.com/gammazero/workerpool"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type BaseVoteCollector struct {
```

Comment:
The `BaseVoteCollector` doesn't actually collect anything, does it? Could we call it a `CollectionBase`? `CollectionIndex`?

That way, it's clear where the actual "vote payloads" are not.

Reply:
It's a _base_ class for all collectors, base in terms of deriving it will share some common information between collectors

Reply:
I don't have any issue with the string "Base". I'm a bit more worried about "Collector".
I understand you're reusing OO naming conventions, but this is Go embedding. :)